DIPs/DIPNNNN.md

```diff
@@ -0,0 +1,349 @@
+   cannot be addressed by adjusting the code.
+
+5. Any form of static reflection that implies access to specific named symbol,
+   for example, `__traits(getMember)`, should still print the deprecation
+   message.
+
+As first implementation step this functionality should only be available as a
+hard list of symbols built into compiler itself and reserved for druntime
+usage. See [breaking changes section](#breaking-changes--deprecation-process)
+for detailed reasoning.
+
+Once concept proves itself with druntime, it should be proposed as an actual
+language feature for mass usage - by either updating this DIP or proposing new
+one. For example, one can simply expose it with new attribute, `@future`.
+
+### Rationale
+
+1. Currently there is no way to warn users of a library about planned addition
+   of new symbol.
+2. Existing workarounds don't eliminate risk of breakage completely and all
+   suffer from some practical drawbacks.
+3. Described problems make it very hard to add new symbols to
+   `object.d` because it is implicitly imported in all D module of all D
+   projects, increasing the chance of name clash as D userbase grows.
+4. Proposed change is not argued to be of notable importance for many D
+   projects. Instead, it is proposed on the basis of being very important for
+   few most widely used one, making most of its impact transitively.
+5. Having simple and reliable way to add new symbols to runtime will reduce
+   amount of time spent in naming debates and improve readability of resulting
+   APIs.
+6. Attempt to introduce proposed functionality is not a one way ticket because
+   it is intended to not be exposed for public usage initially and thus can
+   be judged (and possibly reverted) again when the time comes.
+
+As an extra benefit, such facility can provide a deprecation path for
+introducing new language keywords if the need arises, though this use case is
+beyond the scope of this DIP and won't be evaluated in details.
+
+### Comparison against other languages
+
+Existing programming languages can be separated in 3 main groups in regards
+of this issue:
+
+#### With single global namespace
+
+Examples: C, early PHP
+
+Such languages tend to rely on [using obscure names](#using-more-obscure-name)
+that makes symbol clash less likely but in general problem is still present.
+When the problem manifests, situation becomes much worse than in D, because such
+languages lack tools to disambiguate name clash explicitly.
+
+#### With namespaces, allowing non-qualified access
+
+Examples: C++, C#
+
+This category is most simple to D, meaning that language provides tools to
+disambiguate any names but at the same time allows imports with non-qualified
+access to imported symbols.
+
+In D this is default `import` behavior.
+
+In C++ : `using namespace Namespace.Subnamespace`.
+
+In C# : `using Namespace.Subnamespace`
+
+Importance of problem varies depending on idiomatic code style dominant for the
```

Comment:
s/Importance/The importance/
s/idiomatic code style dominant/the prevalent idioms used/